lib/src/client.dart

```diff
@@ -1189,21 +1178,25 @@ class Client extends MatrixApi {
 
   // TODO: remove once we are able to autogen this
   Future<ServerConfig> _getAuthenticatedConfig() async {
-    String path;
-    if (await authenticatedMediaSupported()) {
-      path = '_matrix/client/v1/media/config';
-    } else {
-      path = '_matrix/media/v3/config';
-    }
-    final requestUri = Uri(path: path);
-    final request = Request('GET', baseUri!.resolveUri(requestUri));
-    request.headers['authorization'] = 'Bearer ${bearerToken!}';
-    final response = await httpClient.send(request);
-    final responseBody = await response.stream.toBytes();
-    if (response.statusCode != 200) unexpectedResponse(response, responseBody);
-    final responseString = utf8.decode(responseBody);
-    final json = jsonDecode(responseString);
-    return ServerConfig.fromJson(json as Map<String, Object?>);
+    Future<ServerConfig> request(String path) async {
+      final requestUri = Uri(path: path);
+      final request = Request('GET', baseUri!.resolveUri(requestUri));
+      request.headers['authorization'] = 'Bearer ${bearerToken!}';
+      final response = await httpClient.send(request);
+      final responseBody = await response.stream.toBytes();
+      if (response.statusCode != 200) {
+        unexpectedResponse(response, responseBody);
+      }
+      final responseString = utf8.decode(responseBody);
+      final json = jsonDecode(responseString);
+      return ServerConfig.fromJson(json as Map<String, Object?>);
+    }
+
+    try {
+      return request('_matrix/client/v1/media/config');
+    } catch (e) {
```

Comment:
Do we know the exact error message that would be returned here? Then we should listen to only this and rethrow other errors